src/connection/ConnectionOptionsReader.ts

```diff
@@ -11,6 +11,11 @@ import {ConnectionOptionsXmlReader} from "./options-reader/ConnectionOptionsXmlR
  */
 export class ConnectionOptionsReader {
 
+    static findOptionByName(options: ConnectionOptions[], name = "default") {
+        const [targetOptions] = options.filter(options => options && options.name === name || (name === "default" && options && !options.name));
```

Comment:
Why use destructuring + filter instead of `find`?

Reply:
@imnotjames filter does not throw error
![image](https://user-images.githubusercontent.com/53092319/94848409-a0561400-042c-11eb-895b-c722536200b2.png)

Reply:
Right - filter skips over the empty item you've entered into the array but I don't really see that as being a huge benefit when you're updating to defend against an undefined value anyway.  The code in your screenshot is different from the code you've entered in this PR.